File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.java
class Solution {
    List<List<Integer>> res;
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        res = new ArrayList<>();
        Arrays.sort(candidates);
        dfs(candidates, target, new ArrayList<>(), 0, 0);
        return res;
    }


    void dfs(int[] nums, int target, List<Integer> subset, int i, int total){
        if(total == target){
            res.add(new ArrayList<>(subset));
            return;
        }

        if(total > target || i == nums.length){
            return;
        }
        subset.add(nums[i]);
        dfs(nums,target,subset,i+1,nums[i] + total);
        subset.remove(subset.size()-1);

        while(i + 1 < nums.length && nums[i] == nums[i+1]){
            i++;
        }
        dfs(nums, target, subset, i+1, total);

    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {

        HashSet<int> results = new HashSet<int>();

        for (int i=0; i<nums.Length;i++){
            for(int j=0; j<nums.Length;j++){
                if (nums[i] + nums[j] == target){
                    results.Add(i);
                    results.Add(j);
                    i = nums.Length;
                    j = nums.Length;
                }
            }

        }

        return results.ToArray();

    }
}
